languages/emf-metamodel/src/main/java/de/jplag/emf/util/EmfaticModelView.java

```diff
@@ -0,0 +1,195 @@
+package de.jplag.emf.util;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.emf.ecore.EEnumLiteral;
+import org.eclipse.emf.ecore.ENamedElement;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.ETypedElement;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.emf.ecore.util.EcoreUtil.Copier;
+import org.eclipse.emf.emfatic.core.generator.emfatic.Writer;
+
+import de.jplag.Token;
+import de.jplag.emf.MetamodelToken;
+import de.jplag.emf.MetamodelTokenType;
+
+/**
+ * Textual view of an EMF metamodel based on Emfatic.
+ * @author Timur Saglam
+ */
+public final class EmfaticModelView extends AbstractModelView {
```

Comment:
This class contains a lot of "magic" to track the objects in the serialization. Maybe the EMFatic parser could be reused to have AST-like information instead of the raw string? Anyways, for the scope of this PR we can leave it as is, but please add more explaining comments about the assumptions you have regarding the syntax.